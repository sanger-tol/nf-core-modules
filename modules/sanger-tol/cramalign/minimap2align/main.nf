process CRAMALIGN_MINIMAP2ALIGN {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/65/65858e733832166824cfd05291fc456bdf219b02baa3944c2c92efad86a6ee7f/data' :
        'community.wave.seqera.io/library/htslib_minimap2_samtools_gawk_perl:6729620c63652154' }"

    input:
    tuple val(meta), path(cram), path(crai), val(chunkn), val(range), path(reference)

    output:
    tuple val(meta), path("*.bam"), emit: bam
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args1 = task.ext.args1 ?: ''
    def args2 = task.ext.args2 ?: ''
    def args3 = task.ext.args3 ?: ''
    def args5 = task.ext.args5 ?: ''
    def prefix = task.ext.prefix ?: "${cram}.${chunkn}.${meta.id}"
    def post_filter = task.ext.args4 ? "samtools view -h ${task.ext.args4} |" : ''

    """
    samtools cat ${args1} -r "#:${range[0]}-${range[1]}" ${cram} | \\
        samtools fastq ${args2} - |  \\
        minimap2 -t${task.cpus} ${args3} ${reference} - | \\
        ${post_filter} \\
        samtools sort ${args5} -@${task.cpus} -T ${prefix}_sort_tmp -o ${prefix}.bam -

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//' )
        minimap2: \$(minimap2 --version | sed 's/minimap2 //g')
    END_VERSIONS
    """

    stub:
    def prefix  = task.ext.prefix ?: "${cram}.${chunkn}.${meta.id}"
    """
    touch ${prefix}.bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//' )
        minimap2: \$(minimap2 --version | sed 's/minimap2 //g')
    END_VERSIONS
    """
}
