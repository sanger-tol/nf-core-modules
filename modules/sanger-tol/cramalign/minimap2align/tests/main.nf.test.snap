{
    "meles meles - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "PAE35587_pass_1f1f0707_115.subset.cram.1.test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,55891c41bf5000a37ba817c3431dd57b"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        "PAE35587_pass_1f1f0707_115.subset.cram.1.test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,55891c41bf5000a37ba817c3431dd57b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-10-07T16:31:23.252769"
    },
    "Meles meles - PacBio HiFi reads": {
        "content": [
            "dfcc4dbebe23293f4f0b9338c2aef9cd",
            [
                "versions.yml:md5,55891c41bf5000a37ba817c3431dd57b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-10-07T16:31:02.641373"
    },
    "Meles meles - ONT reads": {
        "content": [
            "c2aa5ec574995707a01cb9b4b887b4a9",
            [
                "versions.yml:md5,55891c41bf5000a37ba817c3431dd57b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-10-07T16:33:11.776935"
    }
}